### PR TITLE
Extract helper function from the config pkg

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Moving pure helper function to pkg/config/utils so they can be used with comp/core/config comp

### Describe how to test/QA your changes

Our automatic tests should be enough to detect any regression. The deploy to our staging env during the QA process will also detect any regression.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/forwarder/defaultforwarder/forwarder_test.go [hunk 1]
```diff
@@ -22,15 +22,15 @@ import (
 	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder/endpoints"
 	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder/transaction"
 	"github.com/DataDog/datadog-agent/pkg/config"
-	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
```
Comment (hush-hush) [inline comp/forwarder/defaultforwarder/forwarder_test.go:25]: We were importing the config package twice here.

**Review by vickenty — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by AliDatadog — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by BaptisteFoy — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]
Looks good from APM!

**Pull request merged**